lib/chef/event_dispatch/dispatcher.rb

```diff
@@ -18,6 +18,16 @@ def register(subscriber)
         @subscribers << subscriber
       end
 
+      # Check to see if we are dispatching to a formatter
+      def formatter?
+        @subscribers.each do |s|
+          if s.class <= Chef::Formatters::Base && s.class != Chef::Formatters::NullFormatter
+            return true
+          end
```

Comment:
can we make this a property of the formatter classes themselves rather than this class having this special information about them?